src/screenComponents/mineSweeper.cpp

```diff
@@ -0,0 +1,144 @@
+#include "mineSweeper.h"
+#include "miniGame.h"
+#include "hackingDialog.h"
+#include "gui/gui2_togglebutton.h"
+#include "gui/gui2_label.h"
+#include "gui/gui2_progressbar.h"
+#include "gui/gui2_panel.h"
+
+MineSweeper::MineSweeper(GuiPanel* owner, GuiHackingDialog* parent, int difficulty)
+: MiniGame(owner, parent, difficulty) {
+    for(int x=0; x<difficulty; x++)
```

Comment:
I understand the need for a single difficulty parameter. But I think it would be good to initialize a "hacking_field_size" and "bomb_count" from that, like in the old code. It reads pretty difficult now with "difficulty" being used for various things.

Reply:
+1 for readability. Did that here and for the LightsOut.